manim/renderer/cairo_renderer.py

```diff
@@ -71,12 +38,58 @@ def init_scene(self, scene):
             scene.__class__.__name__,
         )
 
-    @pass_scene_reference
-    @handle_caching_play
-    @handle_play_like_call
     def play(self, scene, *args, **kwargs):
-        if scene.compile_animation_data(*args, **kwargs):
+        # Reset skip_animations to the original state.
+        # Needed when rendering only some animations, ans skipping others.
+        self.skip_animations = self._original_skipping_status
+        self.update_skipping_status()
+
+        scene.compile_animation_data(*args, **kwargs)
+
+        is_cached = False
+        # If skip_animations is already True, we can skip all the caching process.
+        if not config["disable_caching"] and not self.skip_animations:
+            hash_current_animation = get_hash_from_play_call(
+                scene, self.camera, scene.animations, scene.mobjects
+            )
+            if self.file_writer.is_already_cached(hash_current_animation):
+                logger.info(
+                    f"Animation {self.num_plays} : Using cached data (hash : %(hash_current_animation)s)",
+                    {"hash_current_animation": hash_current_animation},
+                )
+                is_cached = True
+        else:
+            hash_current_animation = f"uncached_{self.num_plays:05}"
+
+        if is_cached:
+            self.skip_animations = True
```

Comment:
This can be moved under line 60 since it's only run if is_cached is set to True.

Reply:
You're totally right, thanks !